manifests/init.pp

```diff
@@ -121,9 +121,34 @@
   case $::osfamily {
     'RedHat': {
       include nrsysmond::repo::redhat
+      # https://support.newrelic.com/requests/130089
+      # workaround for RPM that's broken on Cent7+ (or any other distro
+      # that uses shadow-utils >= 4.1.5 )
+      if ( $::operatingsystemmajrelease == '7' or $::operatingsystemmajrelease == 7 ) {
```

Comment:
I believe the issue is specific to SELinux and not just CentOS 7. We have CentOS 7 with shadow-utils 4.1.5 machines and this works just fine.
This should test for the `$::selinux` fact instead of `$::operatingsystemmajrelease`